build_runner/lib/src/build_script_generate/bootstrap.dart

```diff
@@ -187,13 +189,19 @@ final _previousLocationsFile = File(
 /// pre-emptively resolve them by resnapshotting, see
 /// https://github.com/dart-lang/build/issues/1929.
 Future<bool> _checkImportantPackageDeps() async {
-  var currentLocationsContent = _importantPackages
-      .map((pkg) => Isolate.resolvePackageUri(
+  var currentLocationsContent = await Stream.fromIterable(_importantPackages)
+      .asyncMap((pkg) => Isolate.resolvePackageUri(
           Uri(scheme: 'package', path: '$pkg/fake.dart')))
       .join('\n');
```

Comment:
There wasn't any existing testing around this content, so I wasn't sure what to do.

If you'd like me to add a test, just let me know how I should go about this.

Reply:
You could write a test using the `.packages` file to get the expected locations. Not required though.

Reply:
I opted not to since I'm not 100% clear on the best way to go about that; sorry 😕.